bin/light-base/Cargo.toml

```
@@ -21,6 +21,7 @@ itertools = "0.10.3"
 log = { version = "0.4.17", default-features = false }
 lru = { version = "0.7.7", default-features = false }  # TODO: there's no way to use a custom hasher; remove this dependency
 rand = "0.8.5"
+serde = { version = "1.0.137", default-features = false, features = ["alloc", "derive"] }
```

```suggestion
serde = { version = "1.0", default-features = false, features = ["alloc", "derive"] }
```

does it have to be >= 137?

---

Well, I don't know, I might be using features that were added only in `1.0.130` or something, in which case we want to prevent users from trying to compile with a version below that.

In general I don't understand what it would bring to use `1.0` instead of `1.0.137`.

---

> In general I don't understand what it would bring to use 1.0 instead of 1.0.137.

broader scope of supported versions, so other libraries who're dependent on smoldot don't have to upgrade to 137?

You can check if code compiles using https://github.com/taiki-e/cargo-minimal-versions.

Anyway, I don't have strong opinion here (in smoldot).